src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.cxx

```diff
@@ -0,0 +1,745 @@
+        interpolated_surface_params[dim] 
+          = (surface_parameters[0 + dim] 
+          * (1 - ref_coords[(i_faces / 2 + 1) % 3])
+          + surface_parameters[(i_faces / 2 == 1 ? 4 : 2) + dim] 
+          * (ref_coords[(i_faces / 2 + 1) % 3]))
+          * (1 - ref_coords[(i_faces / 2 + 2) % 3]);
+        interpolated_surface_params[dim] 
+         += (surface_parameters[(i_faces / 2 == 1 ? 2 : 4) + dim] 
+          * (1 - ref_coords[(i_faces / 2 + 1) % 3])
+          + surface_parameters[6 + dim] 
+          * (ref_coords[(i_faces / 2 + 1) % 3]))
+          * (ref_coords[(i_faces / 2 + 2) % 3]);
+        /* Correct the interpolated parameters with the parameter displacement generated by the edges */
+        interpolated_surface_params[dim] +=
+          surface_parameter_displacement_from_edges[dim];
+      }
+
+      /* Retrieve the surface of the edge */
+      T8_ASSERT (faces[i_faces] <= occ_shape_face_map.Size ());
+      surface =
+        BRep_Tool::Surface (TopoDS::Face (occ_shape_face_map.FindKey (faces[i_faces])));
+
+      /* Check if surface is valid */
+      T8_ASSERT (!surface.IsNull ());
+
+      /* Compute point on surface with interpolated parameters */
+      surface->D0 (interpolated_surface_params[0],
+                   interpolated_surface_params[1], pnt);
+
+      /* Compute the displacement between surface and interpolated coords, scale them with the appropriate ref_coord 
+       * and add them to the out_coords. */
+      if (i_faces % 2 == 0) {
+        out_coords[0] 
+         += (pnt.X () - interpolated_face_coords[0]) 
+          * (1 - ref_coords[i_faces / 2]);
+        out_coords[1] 
+         += (pnt.Y () - interpolated_face_coords[1]) 
+          * (1 - ref_coords[i_faces / 2]);
+        out_coords[2] 
+         += (pnt.Z () - interpolated_face_coords[2]) 
+          * (1 - ref_coords[i_faces / 2]);
+      }
+      else {
+        out_coords[0] 
+         += (pnt.X () - interpolated_face_coords[0]) 
+          * ref_coords[i_faces / 2];
+        out_coords[1] 
+         += (pnt.Y () - interpolated_face_coords[1]) 
+          * ref_coords[i_faces / 2];
+        out_coords[2] 
+         += (pnt.Z () - interpolated_face_coords[2]) 
+          * ref_coords[i_faces / 2];
+      }
+      /* *INDENT-ON* */
+    }
+  }
+}
+
+void
+t8_geometry_occ::t8_geom_evalute_jacobian (t8_cmesh_t cmesh,
+                                           t8_gloidx_t gtreeid,
+                                           const double
+                                           *ref_coords,
+                                           double *jacobian_out) const
+{
+  double              h = 1e-9;
+  double              in1[3], in2[3];
+  double              out1[3], out2[3];
+  for (int dim = 0; dim < 3; ++dim) {
+    in1[0] = ref_coords[dim];
+    in1[1] = ref_coords[dim];
+    in1[2] = ref_coords[dim];
+    in2[0] = ref_coords[dim];
+    in2[1] = ref_coords[dim];
+    in2[2] = ref_coords[dim];
```

Comment:
This computation does not seem correct.
Can you please check it and if you think it is correct, explain it?

If the point is X = (1,2,3), then why are you setting in1 and in2 to
(1,1,1)
(2,2,2)
(3,3,3)
?

Reply:
You are right, I think I fixed it in da81c990